kodi_addon_checker/check_repo.py

```diff
@@ -22,6 +24,9 @@ def check_repo(repo_path, parameters):
     else:
         toplevel_folders = sorted(parameters)
 
+    branch_url = ROOT_URL.format(branch=os.environ.get("TRAVIS_BRANCH"))
```

Comment:
Please don't use `os.environ.get("TRAVIS_BRANCH")` anywhere. Read it from the params or the config file. Probably both would be good.

Reply:
ok so we can have a parameter `branch` that will have the same functionality as `TRAVIS_BRANCH`. And for default we can set it's value to None.

Reply:
And what do you do on `None`?

Reply:
yeah that's what I was thinking. Maybe  just not run those check that are dependent on `branch` parameters or  maybe instead of having `None` as default we can have `leia` or  `krypton` as default (assuming most of the add-on will be submitted to those because they are new) ?

Reply:
We could also read the addon.xml and parse the python version to figure out the branch? But a warning, this will only work for single addons. 

See this list https://kodi.wiki/view/Addon.xml#Dependency_versions, but as it turns out it's not a 100% as krypton and leia has the same target here.

Feels like we should make kodi version a needed param. That would also enable us to move most of .tests-config.json into addon-check and just map it to the versions.

Reply:
@Razzeee yes this can be done but I think instead of taking the kodi version as parameter and then changing it into a branch name (for getting addons.xml), we can just have a branch name as parameter.  This will save us some code for converting version into branch name.

And about  `.tests-config.json`, it currently doesn't have any code that actually depends on version.
 For example: this is a config file from branch `leia`
```json
{
    "check_legacy_language_path": true,
    "check_legacy_strings_xml": true,
    "check_license_file_exists": true,
    "check_kodi_leia_deprecations": true
}
```
Almost all the branch have similar config file and here none of the function directly depends upon the version. So might not have a huge impact on config but yes it will be helpful in future.

Reply:
sorry, that's what I ment by kodi version. As the should match branch names. SO we should be fine.

`.tests-config.json` is completly version dependent, what you state is wrong. It's pretty much the whole reason to even have it. The `check_kodi_leia_deprecations` option is pretty clearly refering to every branch leia+. Please look again and let me know if you have problems understanding :)

Reply:
ok so I'll just add an argument for it.

I had a look to what `check_kodi_leia_deprecation` and what I am able to understand is that it check for strings that were present in old add-ons but are not supposed to be in any add-on submitted to Leia right ?

Reply:
To leia or newer, yes.
The other config items basically work along the same lines. Usually some version dependent changes to kodi engine.